hikari/impl/entity_factory.py

```diff
@@ -2532,28 +2499,35 @@ def _deserialize_interaction_member(
             guild_flags=guild_flags,
         )
 
+    def _deserialize_interaction_channel(
+        self, payload: data_binding.JSONObject
+    ) -> base_interactions.InteractionChannel:
+        channel_id = snowflakes.Snowflake(payload["id"])
+
+        thread_metadata = (
+            self._deserialize_thread_metadata(payload["thread_metadata"]) if "thread_metadata" in payload else None
+        )
+
+        return base_interactions.InteractionChannel(
```

Comment:
what about the other fields? while the documentation for resolved channels is likely just out of date in its list of included fields, the documentation for `interaction.channels` doesn't specify fields which would be included and there's just fields (including channel type specific fields) being missed here for that

Reply:
Bump

Reply:
For now, I will keep it at just resolving the fields that are documented. If Discord document more in the future, this can be extended. Otherwise, we will have a mess of objects because they might be different in different circumstances that are not documented.

If you spot any fields that are consistency sent, we can try and get it documented in the docs and add it to hikari

Reply:
interaction.channel does not have a specific set of fields documented for it, that only applies to resolved options

Reply:
would be nice for interaction.channel to not miss fields